splinterd/src/config/mod.rs

```diff
@@ -134,3 +134,305 @@ impl Config {
         self.heartbeat_interval
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use std::fs;
+
+    /// Paths to existing example config toml files from the top-level Splinterd directory.
+    static TEST_TOML1: &str = "sample_configs/splinterd.toml.example";
+    /// This toml file is only used in testing the TomlConfig module and therefore only used with
+    /// the `toml-config` feature.
+    #[allow(dead_code)]
+    static TEST_TOML2: &str = "sample_configs/splinterd.toml.example2";
+
+    /// Values present in the existing example config toml files.
+    static STORAGE: &str = "yaml";
+    static TRANSPORT: &str = "tls";
+    static CA_CERTS: &str = "certs/ca.pem";
+    static CLIENT_CERT: &str = "certs/client.crt";
+    static CLIENT_KEY: &str = "certs/client.key";
+    static SERVER_CERT: &str = "certs/server.crt";
+    static SERVER_KEY: &str = "certs/server.key";
+
+    /// Config values unique to the TEST_TOML1 file.
+    static SERVICE_ENDPOINT: &str = "127.0.0.1:8043";
+    static NETWORK_ENDPOINT: &str = "127.0.0.1:8044";
+    static NODE_ID: &str = "012";
+
+    /// Config values unique to the TEST_TOML2 file.
+    /// These values are only used in testing the TomlConfig module and therefore only used with
+    /// the `toml-config` feature.
+    #[allow(dead_code)]
+    static SERVICE_ENDPOINT2: &str = "127.0.0.1:8045";
+    #[allow(dead_code)]
+    static NETWORK_ENDPOINT2: &str = "127.0.0.1:8046";
+    #[allow(dead_code)]
+    static NODE_ID2: &str = "345";
+
+    /// Creates a Config struct based on the values from the TEST_TOML1 file.
+    fn construct_config_example() -> Config {
+        Config {
+            storage: Some(STORAGE.to_string()),
+            transport: Some(TRANSPORT.to_string()),
+            cert_dir: None,
+            ca_certs: Some(CA_CERTS.to_string()),
+            client_cert: Some(CLIENT_CERT.to_string()),
+            client_key: Some(CLIENT_KEY.to_string()),
+            server_cert: Some(SERVER_CERT.to_string()),
+            server_key: Some(SERVER_KEY.to_string()),
+            service_endpoint: Some(SERVICE_ENDPOINT.to_string()),
+            network_endpoint: Some(NETWORK_ENDPOINT.to_string()),
+            peers: Some(vec![]),
+            node_id: Some(NODE_ID.to_string()),
+            bind: None,
+            #[cfg(feature = "database")]
+            database: None,
+            registry_backend: None,
+            registry_file: None,
+            heartbeat_interval: None,
+        }
+    }
+
+    /// Creates a Config struct based on the values from the TEST_TOML2 file.
+    /// This function is only used in testing the TomlConfig module and therefore only used with
+    /// the `toml-config` feature.
+    #[allow(dead_code)]
```

Comment:
This should be protected with the toml-config feature then, instead of marked dead-code.

Reply:
Addressed.